tests/integration/ambient/baseline_test.go

```diff
@@ -1441,13 +1606,109 @@ spec:
 							from.CallOrFail(t, echo.CallOptions{
 								Address: "111.111.222.222",
 								Port:    to.PortForName("http"),
+								// we want to make sure the workload selector labels are honored, so we assert the response contains either uncaptured-v1 or uncaptured-v2
+								// sample response:
+								//
+								// ServiceVersion=v1
+								// ServicePort=8080
+								// Host=serviceentry.istio.io
+								// URL=/any/path
+								// Cluster=cluster-0
+								// IstioVersion=
+								// Method=GET
+								// Proto=HTTP/1.1
+								// IP=10.244.2.20
+								// Alpn=
+								// RequestHeader=Accept:*/*
+								// RequestHeader=User-Agent:curl/7.81.0
+								// Hostname=uncaptured-v1-868c9b59b5-rxvfq
+								Check: check.BodyContains(`Hostname=uncaptured-v`),
 							})
 						})
 				})
 			}
 		})
 }
 
+// This was successfully smoke tested locally by looking at ztunnel config dump
+// to query the auto-assigned VIP behind the hostname.
+func TestServiceEntryDNSWithAutoAssign(t *testing.T) {
+	framework.NewTest(t).
+		Features("traffic.ambient").
+		Run(func(t framework.TestContext) {
+			t.Skip("this will work if we set DNS_AUTO_ALLOCATE=true and once we have https://github.com/istio/ztunnel/pull/536")
```

Comment:
now that this PR has merged we should be able to get the hostname resolved trivially (maybe a small CNI change to ensure DNS traffic is captured by ztunnel?)

Reply:
tested this locally. it worked with some minor changes to ztunnel, is sort of blocked by https://github.com/istio/ztunnel/issues/582 or other ztunnel work for now